pkgs/tools/security/enum4linux/default.nix

```diff
@@ -0,0 +1,34 @@
+{ stdenv, fetchurl, samba, perl }:
+
+stdenv.mkDerivation rec {
+  pname = "enum4linux";
+  version = "0.8.9";
+  src = fetchurl {
+    url = https://labs.portcullis.co.uk/download/enum4linux-0.8.9.tar.gz;
+    sha256 = "41334df0cb1ba82db9e3212981340372bb355a8160073331d2a1610908a62d85";
+  };
+
+  phases = "unpackPhase installPhase fixupPhase";
+
+  buildInputs = [samba perl];
+
+  unpackPhase = ''
+    tar xzf ${src}
+  '';
+
+  installPhase = ''
```

Comment:
This is how I might write my `installPhase`:

```
  installPhase = ''
    mkdir -p $out/bin
    cp enum4linux.pl $out/bin/

    wrapProgram $out/bin/enum4linux.pl \
      --prefix PATH : ${stdenv.lib.makeBinPath [ openldap samba ]}
  '';
```

The problem with the `enum4linux.pl` script which your package produces, the `enum4linux.pl` which ends up in the users `bin` folder is that it doesn't have `samba` (or `openldap`) in the `PATH`. If you tested this tool and it worked for you it is because you tested it in an environment where you had already added `samba` into your `PATH`, which isn't reproducible. By including the `wrapProgram` with `makeBinPath` we ensure that `enum4linux.pl` will always have access to `samba` (and `openldap`) in a way that is completely reproducible.